### PR TITLE
Social previews | Add Nextdoor preview

### PR DESCRIPTION
**Pull request by manzoorwanijk**
## Proposed Changes

* This PR adds nextdoor.com preview to Social Previews package

## Testing Instructions

- Goto `/posts/:site`
- Click on actions menu (3 dots) for any post
- Click on _"Preview"_
- Confirm that you DO NOT see the Nextdoor section in the sidebar
- Add the blog sticker to your site:
`add_blog_sticker( 'jetpack-social-nextdoor-connection', null, null, <YOUR_SITE_ID> );` 
- Goto `/marketing/connections/:site` and add a Nextdoor connection for the site
- Goto `/posts/:site`
- Click on actions menu (3 dots) for any post
- Click on _"Preview"_
- Confirm that you see the Nextdoor section in the sidebar
- Click on Nextdoor
- Confirm that you see the Nextdoor post preview
- Confirm that the preview matches the link previews on nextdoor.com
- Confirm the preview for different scenarios for a post
    * with and without featured image
    * with featured image and custom media
    * without featured image but with custom media
    * with video as custom media

In Calypso
Screenshot 2023-11-01 at 12 33 58 PM

In Jetpack

- With featured image and video as custom media
    Screenshot 2023-11-01 at 11 15 50 AM
- With featured image and photo as custom media
    Screenshot 2023-11-01 at 11 17 36 AM
- With featured image and photo as custom media but "Share as a social post" disabled
    Screenshot 2023-11-01 at 11 18 31 AM
- With featured image but without custom media
    Screenshot 2023-11-01 at 11 19 04 AM
- Without featured image and without custom media
    Screenshot 2023-11-01 at 11 51 14 AM
- Without featured image and with video as custom media
    Screenshot 2023-11-01 at 12 07 36 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/share/nextdoor-share-preview/index.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { NextdoorPreviews } from '@automattic/social-previews';
+import { PureComponent } from 'react';
+import { decodeEntities } from 'calypso/lib/formatting';
+
+export class NextdoorSharePreview extends PureComponent {
```
Comment (pablinos) [inline client/components/share/nextdoor-share-preview/index.jsx:5]: Is this for consistency or is there an advantage to this over a functional component?

File: packages/social-previews/src/nextdoor-preview/previews.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+						{ __(
+							'This is what it will look like when someone shares the link to your WordPress post on Nextdoor.',
+							'social-previews'
+						) }
+						&nbsp;
+						<ExternalLink href="https://jetpack.com/support/jetpack-social-image-generator">
+							{ __( 'Learn more about links', 'social-previews' ) }
+						</ExternalLink>
+					</p>
+					<NextdoorLinkPreview { ...props } name="" profileImage="" />
```
Comment (pablinos) [inline packages/social-previews/src/nextdoor-preview/previews.tsx:48]: I'm not sure if this problem is isolated to Nextdoor, but we're using the selected media for the link preview when a featured image is set on the post. I'm not sure that's correct, as I think the featured image will be used as the Open Graph image, unless Social Image Generator is enabled.

**Review by pablinos — COMMENTED** [review]
This is great, it's working well for me. Just a couple of queries.

File: client/components/share/nextdoor-share-preview/index.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/components/share/nextdoor-share-preview/index.jsx:5]: Just for consistency.

File: packages/social-previews/src/nextdoor-preview/previews.tsx (hunk 2 above)
Comment (manzoorwanijk) [reply packages/social-previews/src/nextdoor-preview/previews.tsx:48]: IIRC, I asked this question before probably in a team meeting whether the custom media is used as OG image and the answer was "Yes".

Comment (manzoorwanijk) [reply packages/social-previews/src/nextdoor-preview/previews.tsx:48]: Also, if it's a problem, then it's with all the reviews then and needs a fix in Jetpack, not here.

Comment (pablinos) [reply packages/social-previews/src/nextdoor-preview/previews.tsx:48]: Sounds like it is working the right way! p1698924866197229/1698825180.539239-slack-C02JJ910CNL

**Review by pablinos — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9582053

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @manzoorwanijk for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**